to-do-list/src/App.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState, useMemo, useEffect, useCallback } from 'react';
+import '../src/components/UI//styles/App.css'
+import { PostItemProps } from './components/Interfaces/interfaces';
+import PostList from './components/PostList/PostList';
+import PostForm from './components/PostForm/PostForm';
+import SearchBar from './components/SearchBar/SearchBar';
+import MyModal from './components/UI/modal/MyModal';
+import PostService from './API/PostService';
+import Weather from './components/Weather/Weather';
+
+
+function App() {
+  
+  const [posts, setPosts] = useState<PostItemProps[]>([])
+  const [filter, setFilter] = useState({query: ''})
+  const [modal, setModal] = useState(false)
+
+  const sortedAndSearchedPosts = useMemo(()=>{
+    return posts.filter(post => post.title.toLowerCase().includes(filter.query))
+
+  }, [filter.query, posts])
+
+  useEffect(() => {
+    let canceled = false;
+  
+    async function fetchPosts() {
+      const posts = await PostService.getAll();
+      if (!canceled) {
+        setPosts(posts.tasks);
+      }
+    }
+  
+    fetchPosts();
+  
+    return () => {
+      canceled = true;
+    };
+  }, []);
+  
+
+  const createPost = useCallback(async (newPost: PostItemProps) => {
+    const createdPost = await PostService.create(newPost);
+    setPosts(posts => [...posts, createdPost])
+    setModal(false)
+  }, [])
+
+  const removePost = useCallback(async (id: number) => {
+    await PostService.delete(id);
+    setPosts(posts => posts.filter(p => p.id !== id));
+  }, [setPosts]);
+  
+  const closeModal = useCallback(() => {
+    setModal(false);
+  }, []);
+
+
+  return (
+    <div className="container">
+      <MyModal visible={modal} setVisible={closeModal}>
```

Comment:
Don't use names like this, use meaningful names like CreateTaskModal
Also, setVisible sounds like you can both show and hide modal with this method. It's better to call closeModal - call methods and variables according to their actions